sys/vfs/vfs.c

```diff
@@ -727,7 +728,8 @@ int vfs_bind(int fd, int flags, const vfs_file_ops_t *f_op, void *private_data)
 
 int vfs_normalize_path(char *buf, const char *path, size_t buflen)
 {
-    DEBUG("vfs_normalize_path: %p, \"%s\" (%p), %lu\n", buf, path, path, (unsigned long)buflen);
+    DEBUG("vfs_normalize_path: %p, \"%s\" (%p), %lu\n",
+          (void *)buf, path, (void *)path, (unsigned long)buflen);
```

Comment:
You didn't make that change but I was just wondering if there is any reason why `%zu` is not used to print the size_t buflen value instead of casting it to an `unsigned long`.

The `fprintf(3)` man page states the following:

```
       z       Specifies that a following d, i, o, u, x, or X conversion
               specifier applies to a size_t or the corresponding signed
               integer type argument; or that a following n conversion
               specifier applies to a pointer to a signed integer type
               corresponding to a size_t argument.
```

Reply:
IIRC it wasn't working when using newlib nano printf, therefore we cast to unsigned long instead.

Reply:
@gebart is newlib nano enabled by default?

Reply:
Apparently changing this to `%zu` doesn't affect at all and IMHO is better. @smlng can you try?

Reply:
Did you test this with newlib nano or only with newlib?

Reply:
Yes, cortex m use newlib nano by default when it is available

Reply:
Well I made the test with some cortex platforms and everything went ok... Anyways, as said below by @smlng this is not the point of the PR and I think it can be improved later.